pkgs/by-name/pl/plex-desktop/package.nix

```diff
@@ -104,48 +105,52 @@ let
-        rm $out/usr/lib/x86_64-linux-gnu/libasound.so.2
-        ln -s ${alsa-lib}/lib/libasound.so.2 $out/usr/lib/x86_64-linux-gnu/libasound.so.2
-        rm $out/usr/lib/x86_64-linux-gnu/libasound.so.2.0.0
-        ln -s ${alsa-lib}/lib/libasound.so.2.0.0 $out/usr/lib/x86_64-linux-gnu/libasound.so.2.0.0
+      ln -s ${libedit}/lib/libedit.so.0 $out/lib/libedit.so.2
+      rm $out/usr/lib/x86_64-linux-gnu/libasound.so.2
+      ln -s ${alsa-lib}/lib/libasound.so.2 $out/usr/lib/x86_64-linux-gnu/libasound.so.2
+      rm $out/usr/lib/x86_64-linux-gnu/libasound.so.2.0.0
+      ln -s ${alsa-lib}/lib/libasound.so.2.0.0 $out/usr/lib/x86_64-linux-gnu/libasound.so.2.0.0
 
-        runHook postInstall
-      '';
+      runHook postInstall
+    '';
   };
 in
 buildFHSEnv {
-    name = "${pname}-${version}";
-    targetPkgs = pkgs: [ xkeyboard_config ];
+  inherit pname version meta;
+  targetPkgs = pkgs: [ xkeyboard_config ];
 
-    extraInstallCommands = ''
-      mkdir -p $out/share/applications $out/share/icons/hicolor/scalable/apps
-      install -m 444 -D ${plex-desktop}/meta/gui/plex-desktop.desktop $out/share/applications/plex-desktop.desktop
-      substituteInPlace $out/share/applications/plex-desktop.desktop \
-        --replace-fail \
-        'Icon=''${SNAP}/meta/gui/icon.png' \
-        'Icon=${plex-desktop}/meta/gui/icon.png' \
-        --replace-fail \
-        'Exec=plex-desktop' \
-        'Exec=plex-desktop-${version}'
-    '';
+  extraInstallCommands = ''
+    mkdir -p $out/share/applications $out/share/icons/hicolor/scalable/apps
+    install -m 444 -D ${plex-desktop}/meta/gui/plex-desktop.desktop $out/share/applications/plex-desktop.desktop
+    substituteInPlace $out/share/applications/plex-desktop.desktop \
+      --replace-fail \
+      'Icon=''${SNAP}/meta/gui/icon.png' \
+      'Icon=${plex-desktop}/meta/gui/icon.png' \
+      --replace-fail \
+      'Exec=plex-desktop' \
+      'Exec=plex-desktop-${version}'
+  '';
 
-    runScript = writeShellScript "plex-desktop.sh" ''
-      # Widevine won't download unless this directory exists.
-      mkdir -p $HOME/.cache/plex/
-      PLEX_USR_PATH=${lib.makeSearchPath "usr/lib/x86_64-linux-gnu"  [ plex-desktop ]}
+  runScript = writeShellScript "plex-desktop.sh" ''
+    # Widevine won't download unless this directory exists.
+    mkdir -p $HOME/.cache/plex/
+    PLEX_USR_PATH=${lib.makeSearchPath "usr/lib/x86_64-linux-gnu" [ plex-desktop ]}
 
-      set -o allexport
-      LD_LIBRARY_PATH=${lib.makeLibraryPath [ plex-desktop libglvnd-1_4_0 ]}:$PLEX_USR_PATH
-      LIBGL_DRIVERS_PATH=$PLEX_USR_PATH/dri
-      ${lib.toShellVars extraEnv}
-      exec ${plex-desktop}/Plex.sh
+    set -o allexport
+    LD_LIBRARY_PATH=${
+      lib.makeLibraryPath [
+        plex-desktop
+        libglvnd-1_4_0
+      ]
+    }:$PLEX_USR_PATH
+    LIBGL_DRIVERS_PATH=$PLEX_USR_PATH/dri
+    ${lib.toShellVars extraEnv}
+    exec ${plex-desktop}/Plex.sh
   '';
+  passthru.updateScript = ./update.sh;
```

Comment:
have you tested that this script works? just wanted to be sure since it seems that this file has existed for a while but was never used in the previous drv

Reply:
Thanks or checking, I think I needed to add  `export NIXPKGS_ALLOW_UNFREE=1`just before the build step.

With that added, `nix-update plex-desktop --use-update-script` (or just running the script) produces the commit `plex-desktop: 1.101.0 -> 1.102.0` with the correct changes. It's pretty much a carbon copy of  [spotify's update](https://github.com/NixOS/nixpkgs/blob/ce76e7aa78ace4a8cc96582f6d9d2b12eb7606a1/pkgs/applications/audio/spotify/update.sh) script.

Reply:
awesome, thanks!